tests/robots/test_targets.py

```diff
@@ -0,0 +1,224 @@
+    nt = ConstraintSetTarget.__from_data__(target.__data__)
+    assert target.constraint_set == nt.constraint_set
+    assert target.name == nt.name
+    for c1, c2 in zip(target.constraint_set, nt.constraint_set):
+        assert c1.__data__ == c2.__data__
+
+
+@pytest.fixture
+def frame_waypoints():
+    target_frames = []
+    target_frames.append(Frame(Point(1.0, -2.0, 3.0), Vector(1.0, 0.0, 0.0), Vector(0.0, 1.0, 0.0)))
+    target_frames.append(Frame(Point(4.0, -5.0, 6.0), Vector(1.0, 0.0, 0.0), Vector(0.0, 1.0, 0.0)))
+    target_frames.append(Frame(Point(7.0, -8.0, 9.0), Vector(-1.0, 0.0, 0.0), Vector(0.0, 1.0, 0.0)))
+    tolerance_position = 0.001
+    tolerance_orientation = 0.001
+    tool_coordinate_frame = Frame(Point(0.0, 10.0, 20.0), Vector(1.0, 0.0, 0.0), Vector(0.0, 1.0, 0.0))
+    name = "my testing waypoints"
+    return FrameWaypoints(target_frames, tolerance_position, tolerance_orientation, tool_coordinate_frame, name)
+
+
+@pytest.fixture
+def point_axis_waypoints():
+    target_points_and_axes = []
+    target_points_and_axes.append((Point(1.0, -2.0, 3.0), Vector(1.0, 0.0, 0.0)))
+    target_points_and_axes.append((Point(4.0, -5.0, 6.0), Vector(1.0, 0.0, 0.0)))
+    target_points_and_axes.append((Point(7.0, -8.0, 9.0), Vector(-1.0, 0.0, 0.0)))
+    tolerance_position = 0.001
+    tool_coordinate_frame = Frame(Point(0.0, 10.0, 20.0), Vector(1.0, 0.0, 0.0), Vector(0.0, 1.0, 0.0))
+    name = "my testing waypoints"
+    return PointAxisWaypoints(target_points_and_axes, tolerance_position, tool_coordinate_frame, name)
+
+
+def test_serialization_waypoints(frame_waypoints, point_axis_waypoints):
+    # FrameWaypoints
+    nt = FrameWaypoints.__from_data__(frame_waypoints.__data__)
+    for f1, f2 in zip(frame_waypoints.target_frames, nt.target_frames):
+        assert f1 == f2
+    assert frame_waypoints.tolerance_position, nt.tolerance_position
+    assert frame_waypoints.tolerance_orientation, nt.tolerance_orientation
+    assert frame_waypoints.tool_coordinate_frame == nt.tool_coordinate_frame
+    assert frame_waypoints.name == nt.name
+
+    # PointAxisWaypoints
+    nt = PointAxisWaypoints.__from_data__(point_axis_waypoints.__data__)
+    for (p1, a1), (p2, a2) in zip(point_axis_waypoints.target_points_and_axes, nt.target_points_and_axes):
+        assert p1 == p2
+        assert a1 == a2
+    assert point_axis_waypoints.tolerance_position == nt.tolerance_position
+    assert point_axis_waypoints.tool_coordinate_frame == nt.tool_coordinate_frame
+    assert point_axis_waypoints.name == nt.name
+
+
+def test_target_scale(frame_target):
+    scale_factor = 0.001
+    nt = frame_target.scaled(scale_factor)
+    assert nt.target_frame == frame_target.target_frame.scaled(scale_factor)
+    assert nt.tolerance_position == frame_target.tolerance_position * scale_factor
+    assert nt.tolerance_orientation == frame_target.tolerance_orientation * scale_factor
```

Comment:
I don't know what happened here. This new file was added in another PR, why does it show up again here? Maybe you'd need to merge main back into this? Otherwise, this point has the bug that we fixed last Friday about scaling the orientation.